### ISSUE TITLE
porting to c++ and using sqlite

### ISSUE THREAD

**Issue by milad6506** [bmwcarit/barefoot#28]
dear all
i want to write a map matching software based on barefoot with c++ that works fine on embedded systems for an IVI module. i will publish it opensource along with its proper openembedded layer.
the thing is that i am trying to choose a map database solution. does make sense to ddirectly use sqlite or should turn to something else?
thank you all in advance

**Comment by smattheis** [comment]
I think every DBMS solution will do. It's a benefit to use a DBMS with spatial extensions (geometry data structures, spatial index, etc.), but it's not really necessary as you could just store geometries as text (WKT). SpatialLite could be an option for you (https://www.gaia-gis.it/fossil/libspatialite/index). For the import of OSM data into the database, I would refer to Osmosis which is used in Barefoot (http://wiki.openstreetmap.org/wiki/Osmosis).

**Comment by milad6506** [comment]
Dear Sebastian,

special thanks for your answer.

the issue is I tried using spatialite. the functions are not as mature as
those postgis and this led to the point where I decided to directly use
SQLite.
but for the very basic requirement of finding the set of ways in a
predefined vicinity of a point, one may needed four consecutive queries to
be executed in order to find the list ways and their corresponding nodes.
and if you want to check the ways tags another querry is needed.

I also tried multithreaded access to the database but because of the shared
database it was not of much help.

so I was thinking if it is possible to find a solution in between!



Regards;

*Milad Hasanvand*

*Connected Car System Designer @ AIRIC-SAIPA*

-------------------------------------------------------------------------------------------

<http://airic-ir.com/>

Automotive Industries Research & Innovation Center (AIRIC)
SAIPA Group

15th Km Karadj Makhsoos Rd., Tehran, Iran.
P.O. Box 13445/967
hasanvand.milad@airic-ir.com
Mobile:   +98 938 510 0152
